_posts/2018-02-19-12-04-NavigationPropertiesOnComplexTypes.md

```diff
@@ -0,0 +1,59 @@
+---
+layout: post
+title: "12.4 Navigation Properties on Complex Types"
+description: "Changes required to support navigation properties on complex types in WebAPI"
+category: "12. Design"
+---
+# NAVIGATION PROPERTIES ON COMPLEX TYPES
+## 1	OVERVIEW
+### 1.1	INTRODUCTION
+In WebAPI, navigation properties were only allowed on entity types. This design will now enable support for navigation properties on complex types as well. 
+### 1.2	PROBLEM STATEMENT
+We will ensure that the requests in the table below are supported.   
+
+|Request	| Template |
+|:-------:|:---------|
+|GET |	{resource-path}?$expand=NavigationProperty |
+|GET |	{resource-path}?$expand=Cast/NavigationProperty |
+|GET |	{resource-path}?$expand=ComplexProperty/Cast/NavigationProperty |
+|GET |	{resource-path}?$expand=ComplexProperty/NavigationProperty |
+|GET | {resource-path}?$expand = ComplexProperty/NestedComplexProperty/NavigationProperty |
+|GET |	{collection-resource-path}/{key}?$expand = ComplexProperty/NavigationProperty |
+|GET | {collection-resource-path}/{key}?$expand = ComplexProp1/NavProp1, ComplexProp2/NavProp2 |
+|GET| {resource-path}?$expand=ComplexProperty/NavigationProperty({query-options}) |
```

Comment:
how about the recursive in the complex type?

for example:

ComplexType (C)
   |--- C (cp)
   |--- E (np)